### ISSUE TITLE
oc deploy shows an incorrect status

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#3367]
Scale up an initial deployment to 10.  oc deploy and oc status display the wrong message.

```
$ oc deploy nodejs-frontend
nodejs-frontend #1 deployed 5 minutes ago - 10 pods reducing to 1
```

**Comment by kargakis** [comment]
@ironcladlou probably has to do with the DesiredReplicasAnnotation. We set it up only once when creating the deployment right?

**Comment by kargakis** [comment]
```
[vagrant@openshiftdev sample-app]$ oc describe rc database-1
Name:		database-1
Image(s):	openshift/mysql-55-centos7:latest
Selector:	deployment=database-1,deploymentconfig=database,name=database
Labels:		openshift.io/deployment-config.name=database,template=application-template-stibuild
Replicas:	10 current / 10 desired
Pods Status:	0 Running / 10 Waiting / 0 Succeeded / 0 Failed
Events:
  FirstSeen				LastSeen			Count	From				SubobjectPath	Reason			Message
  Tue, 30 Jun 2015 13:53:01 +0000	Tue, 30 Jun 2015 13:53:01 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-oahyv
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-2stv9
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-xlupa
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-wq0zu
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-38p58
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-kjhqy
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-zn6vb
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-xvg1a
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-7o19b
  Tue, 30 Jun 2015 13:53:02 +0000	Tue, 30 Jun 2015 13:53:02 +0000	1	{replication-controller }			successfulCreate	Created pod: database-1-vsv6x

[vagrant@openshiftdev sample-app]$ oc get rc database-1 -o yaml
apiVersion: v1
kind: ReplicationController
metadata:
  annotations:
    kubectl.kubernetes.io/desired-replicas: "1"
....
```

**Comment by ironcladlou** [comment]
>@ironcladlou probably has to do with the DesiredReplicasAnnotation. We set it up only once when creating the deployment right?

Yeah... we set the desired replicas annotation to drive the single deployment process, and when that's done, it won't be updated again for that deployment. If you scale the existing deployment, the desired annotation value won't match.

For a complete deployment, the spec replicas value should represent the current desired number (and will match the annotation value right when the deployment is complete, but no guarantees later).

**Comment by smarterclayton** [comment]
That's sufficient to distinguish - we need to easily determine the
difference between complete and running and then pick the right value to
show

On Jun 30, 2015, at 11:40 AM, Dan Mace <notifications@github.com> wrote:

@ironcladlou <https://github.com/ironcladlou> probably has to do with the
DesiredReplicasAnnotation. We set it up only once when creating the
deployment right?

Yeah... we set the desired replicas annotation to drive the single
deployment process, and when that's done, it won't be updated again for
that deployment. If you scale the existing deployment, the desired
annotation value won't match.

For a complete deployment, the spec replicas value should represent the
current desired number (and will match the annotation value right when the
deployment is complete, but no guarantees later).

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/issues/3367#issuecomment-117232457>.
